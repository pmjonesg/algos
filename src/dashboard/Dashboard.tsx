import React, { SyntheticEvent } from "react";
import {
  Typography,
  ExpansionPanel,
  ExpansionPanelActions,
  ExpansionPanelDetails,
  ExpansionPanelSummary
} from "@material-ui/core";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import { withStyles } from "@material-ui/core/styles";

const styles = {
  card: {
    minWidth: 275
  },
  title: {
    fontSize: 14
  }
};

type IState = { expanded: string };
type IProps = { classes: any };

const algos = [
  {
    title: "Sorting and Order Statistics",
    algorithms: [
      {
        title: "Heapsort"
      }
    ]
  },
  { title: "Data Structures" },
  { title: "Advanced Desgin and Analysis Techniques" },
  { title: "Advanced Data Structures" },
  { title: "Graph Algorithms" },
  { title: "Selected Topics" }
];

class Dashboard extends React.Component<IProps> {
  state: IState = { expanded: "" };

  constructor(props: IProps) {
    super(props);
  }

  handleChange = (panel: string) => (event: any, expanded: any) => {
    this.setState({
      expanded: expanded ? panel : ""
    });
  };

  render() {
    const { classes } = this.props;
    const { expanded } = this.state;
    return (
      <React.Fragment>
        {algos.map(algo => (
          <ExpansionPanel
            expanded={expanded === algo.title}
            onChange={this.handleChange(algo.title)}
          >
            <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
              <Typography className={classes.title}>{algo.title}</Typography>
            </ExpansionPanelSummary>
            <ExpansionPanelDetails>
              <Typography className={classes.secondaryHeading}>
                {algo.algorithms && algo.algorithms.length
                  ? algo.algorithms.map(algorithm => {
                      return (
                        <Typography className={classes.secondaryHeading}>
                          {algorithm.title}
                        </Typography>
                      );
                    })
                  : "To be implemented"}
              </Typography>
            </ExpansionPanelDetails>
          </ExpansionPanel>
        ))}
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(Dashboard);
