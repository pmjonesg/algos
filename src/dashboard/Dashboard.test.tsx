import React from 'react';
import { render as rtlRender, RenderResult } from 'react-testing-library';

import Dashboard from './Dashboard';
import { BrowserRouter } from 'react-router-dom';

const render = (ui: React.ReactElement<any>, options?: any): RenderResult => {
  return rtlRender(<BrowserRouter>{ui}</BrowserRouter>, options);
};

it('can render dashboard component', () => {
  const { getByTestId } = render(<Dashboard />);
  expect(getByTestId('dashboard')).toMatchSnapshot();
});
