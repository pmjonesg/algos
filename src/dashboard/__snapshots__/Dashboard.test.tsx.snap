// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render dashboard component 1`] = `
<div
  data-testid="dashboard"
>
  <div
    class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiExpansionPanel-root-3 MuiExpansionPanel-rounded-4"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root-40 MuiExpansionPanelSummary-root-34"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content-38"
      >
        <h5
          class="MuiTypography-root-43 MuiTypography-h5-59 MuiTypography-gutterBottom-70"
        >
          Sorting and Order Statistics
        </h5>
      </div>
      <div
        aria-hidden="true"
        class="MuiButtonBase-root-40 MuiIconButton-root-79 MuiExpansionPanelSummary-expandIcon-39"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label-84"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-85"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-100"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiCollapse-container-94"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper-96"
      >
        <div
          class="MuiCollapse-wrapperInner-97"
        >
          <div
            class="MuiExpansionPanelDetails-root-98 Styled-WithStyles-ExpansionPanelDetails---root-bscilo"
          >
            <a
              href="/algorithms/heapsort"
            >
              <div
                class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 Summary-root-99"
                style="width: 100vw;"
              >
                <p
                  class="MuiTypography-root-43 MuiTypography-body1-52"
                >
                  Heapsort
                </p>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiExpansionPanel-root-3 MuiExpansionPanel-rounded-4"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root-40 MuiExpansionPanelSummary-root-34"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content-38"
      >
        <h5
          class="MuiTypography-root-43 MuiTypography-h5-59 MuiTypography-gutterBottom-70"
        >
          Data Structures
        </h5>
      </div>
      <div
        aria-hidden="true"
        class="MuiButtonBase-root-40 MuiIconButton-root-79 MuiExpansionPanelSummary-expandIcon-39"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label-84"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-85"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-100"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiCollapse-container-94"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper-96"
      >
        <div
          class="MuiCollapse-wrapperInner-97"
        >
          <div
            class="MuiExpansionPanelDetails-root-98 Styled-WithStyles-ExpansionPanelDetails---root-bscilo"
          >
            To be implemented
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiExpansionPanel-root-3 MuiExpansionPanel-rounded-4"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root-40 MuiExpansionPanelSummary-root-34"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content-38"
      >
        <h5
          class="MuiTypography-root-43 MuiTypography-h5-59 MuiTypography-gutterBottom-70"
        >
          Advanced Design and Analysis Techniques
        </h5>
      </div>
      <div
        aria-hidden="true"
        class="MuiButtonBase-root-40 MuiIconButton-root-79 MuiExpansionPanelSummary-expandIcon-39"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label-84"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-85"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-100"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiCollapse-container-94"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper-96"
      >
        <div
          class="MuiCollapse-wrapperInner-97"
        >
          <div
            class="MuiExpansionPanelDetails-root-98 Styled-WithStyles-ExpansionPanelDetails---root-bscilo"
          >
            To be implemented
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiExpansionPanel-root-3 MuiExpansionPanel-rounded-4"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root-40 MuiExpansionPanelSummary-root-34"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content-38"
      >
        <h5
          class="MuiTypography-root-43 MuiTypography-h5-59 MuiTypography-gutterBottom-70"
        >
          Advanced Data Structures
        </h5>
      </div>
      <div
        aria-hidden="true"
        class="MuiButtonBase-root-40 MuiIconButton-root-79 MuiExpansionPanelSummary-expandIcon-39"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label-84"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-85"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-100"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiCollapse-container-94"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper-96"
      >
        <div
          class="MuiCollapse-wrapperInner-97"
        >
          <div
            class="MuiExpansionPanelDetails-root-98 Styled-WithStyles-ExpansionPanelDetails---root-bscilo"
          >
            To be implemented
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiExpansionPanel-root-3 MuiExpansionPanel-rounded-4"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root-40 MuiExpansionPanelSummary-root-34"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content-38"
      >
        <h5
          class="MuiTypography-root-43 MuiTypography-h5-59 MuiTypography-gutterBottom-70"
        >
          Graph Algorithms
        </h5>
      </div>
      <div
        aria-hidden="true"
        class="MuiButtonBase-root-40 MuiIconButton-root-79 MuiExpansionPanelSummary-expandIcon-39"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label-84"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-85"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-100"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiCollapse-container-94"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper-96"
      >
        <div
          class="MuiCollapse-wrapperInner-97"
        >
          <div
            class="MuiExpansionPanelDetails-root-98 Styled-WithStyles-ExpansionPanelDetails---root-bscilo"
          >
            To be implemented
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiExpansionPanel-root-3 MuiExpansionPanel-rounded-4"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root-40 MuiExpansionPanelSummary-root-34"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content-38"
      >
        <h5
          class="MuiTypography-root-43 MuiTypography-h5-59 MuiTypography-gutterBottom-70"
        >
          Selected Topics
        </h5>
      </div>
      <div
        aria-hidden="true"
        class="MuiButtonBase-root-40 MuiIconButton-root-79 MuiExpansionPanelSummary-expandIcon-39"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label-84"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-85"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-100"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiCollapse-container-94"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper-96"
      >
        <div
          class="MuiCollapse-wrapperInner-97"
        >
          <div
            class="MuiExpansionPanelDetails-root-98 Styled-WithStyles-ExpansionPanelDetails---root-bscilo"
          >
            To be implemented
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
